shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Chain.hs

```diff
@@ -372,46 +369,42 @@ chainTransition =
         pure $ ChainState nes'' cs' eta0' etaV' etaC' etaH' lab'
 
 instance
-  ( CryptoClass.Crypto c,
-    DSignable c (OCertSignable c),
-    DSignable c (Hash c EraIndependentTxBody),
-    KESignable c (BHBody c),
-    VRF.Signable (VRF c) Seed
+  ( Era era,
+    ShelleyBased era,
+    STS (CHAIN era),
+    STS (BBODY era)
   ) =>
-  Embed (BBODY (ShelleyEra c)) (CHAIN (ShelleyEra c))
+  Embed (BBODY era) (CHAIN era)
```

Comment:
Is this right?

Note that the `STS (CHAIN era)` instance above has a `Embed (BBODY era) (CHAIN era)` constraint, i.e., this instance. Since this instance has a `STS (CHAIN era)` constraint, there is a cycle.

Embed (BBODY era) (CHAIN era),